src/app/modules/pokemon/components/pokemon-card-chart/pokemon-card-chart.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, Input } from '@angular/core';
+import { ChartDataSets, ChartOptions, ChartType } from 'chart.js';
+import { Label } from 'ng2-charts';
+import { Pokemon } from '../../models/pokemon';
+import { barChartOptions as chartOptions } from 'src/app/utils/chart/bar-chart';
+import { barChartPrimaryStyle, barChartSecondaryStyle } from 'src/app/utils/chart/bar-chart';
+
+@Component({
+  selector: 'app-pokemon-card-chart',
+  templateUrl: './pokemon-card-chart.component.html',
+  styleUrls: ['./pokemon-card-chart.component.css']
+})
+export class PokemonCardChartComponent {
+  @Input() isComparing: boolean;
+  @Input()
+  get currentPokemon(): Pokemon {
+    return this.currentPokemonInformation;
+  }
+
+  set currentPokemon(currentPokemon: Pokemon) {
+    if (currentPokemon) {
+      this.barChartLabels = currentPokemon.stats.map((stat) => stat.stat.name);
+      const currentPokemonBaseStat = currentPokemon.stats.map((stat) => stat.base_stat);
+      this.barChartData = [{
+        data: currentPokemonBaseStat,
+        label: currentPokemon.name,
+        ...barChartPrimaryStyle }];
+    }
```

Comment:
this part of the code looks like is shared between the two setters, move it to a function with the if requiring both values